sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/ChallengeBasedAuthenticationPolicyTests.cs

```diff
@@ -76,23 +80,59 @@ public async Task MultipleRequests(int numberOfRequests, int minDelay, int maxDe
             };
 
             using AzureEventSourceListener logger = AzureEventSourceListener.CreateTraceLogger(EventLevel.Verbose);
-            SecretClient client = new SecretClient(s_vaultUri, new MockCredential(transport), options);
+            SecretClient client = new SecretClient(VaultUri, new MockCredential(transport), options);
 
             Task<Response<KeyVaultSecret>>[] tasks = new Task<Response<KeyVaultSecret>>[numberOfRequests];
             for (int i = 0; i < tasks.Length; ++i)
             {
-                tasks[i] = client.GetSecretAsync("test-secret");
+                tasks[i] = Task.Run(async () => await client.GetSecretAsync("test-secret").ConfigureAwait(false));
             }
 
-            await Task.WhenAll(tasks).ConfigureAwait(false);
-
-            for (int i = 0; i < tasks.Length; ++i)
+            foreach (KeyVaultSecret secret in await Task.WhenAll(tasks))
             {
-                KeyVaultSecret secret = await tasks[i].ConfigureAwait(false);
                 Assert.AreEqual("secret-value", secret.Value);
             }
         }
 
+        [Test]
+        public async Task TenantChangedRequest()
+        {
+            MockTransportBuilder builder = new MockTransportBuilder
+            {
+                AccessTokenLifetime = TimeSpan.Zero,
+            };
+            MockTransport transport = builder.Build();
+
+            SecretClientOptions options = new SecretClientOptions
+            {
+                Transport = transport,
+                Diagnostics =
```

Comment:
Remove debugging related stuff.

Reply:
Why? It's useful when debugging next time. All secret information is fake by design. I see no value in removing it.

Reply:
What value is there in keeping it? Should we add it to every client we create in case we need to debug later?

Reply:
To me it's similar to leaving Console.WriteLine everywhere just in case that never get cleaned up.

Reply:
That's oversimplifying it. Concurrency issues are difficult enough to diagnose, which this whole issue is about from the start. But the multiple hops against multiple endpoints makes this even harder. Unless you feel very strongly about this, I prefer to leave it in.

For tests in other public and private repos, yes, I tend to leave stuff in that makes it easier to diagnose test failures. Why remove it if you need it again? It's there when needed, and only happens when you debug tests and only with fake data in this particular case (no PII leaks).

Reply:
Chances are this test would never fail again or if it does it would be completely different issue that would require more/different logging.

I feel strongly about consistency and don't want to encourage people leaving diagnostics enabled on per test basis. I'm open to having a common solution/convention for logging in tests but am against one offs.

Reply:
I'm a fan of consistency as well, but I feel it's not always required. I will remove it from all but the main concurrent method, and I'll comment it out so it's not left running in tests until needed. It should be easy to figure out how to restore what I did in my TDD changes to repro and track down the issue. Yes, hopefully they don't fail again and that's why we have tests, but if they do it's more or less ready to go.